### PR TITLE
Bump github.com/gardener/gardener from 1.75.0 to 1.77.0

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement
/platform aws

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Follow Up - https://github.com/gardener/gardener/pull/8242

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by kon-angelo** [comment]
@acumino - can we do v1.77 directly instead ?

**Comment by acumino** [comment]
> @acumino - can we do v1.77 directly instead ?

Updated it to 1.77. PTAL

**Comment by shafeeqes** [comment]
You could also do https://github.com/gardener/gardener-extension-networking-calico/pull/293/commits/f4b8a4a33665673d3763f39097246044396a9fee

**Review by shafeeqes — DISMISSED** [review]
/lgtm

File: hack/tools.go [hunk 1]
```diff
@@ -26,7 +26,6 @@ import (
 	_ "github.com/gardener/gardener/hack/api-reference/template"
 
 	_ "github.com/ahmetb/gen-crd-api-reference-docs"
-	_ "github.com/golang/mock/mockgen"
```
Comment (ialidzhikov) [inline hack/tools.go:29]: I see the extension needs the `mockgen` binary as it generates mocks. See https://github.com/gardener/gardener-extension-provider-aws/blob/0a7ca2db2e57b2106f239c816b420c6a62395cb3/pkg/aws/client/mock/doc.go#L15.

How the mockgen binary get's now built under `hack/tools/bin`? Is g/g's hack/tools.go get used?

I was thinking that when a project needs such a binary, the needed import needs to be added such as:
```
	_ "go.uber.org/mock/mockgen"
```

Comment (acumino) [reply hack/tools.go:29]: > How the mockgen binary get's now built under hack/tools/bin? Is g/g's hack/tools.go get used?

Mostly yes.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**